src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertySerializer.cs

```diff
@@ -164,17 +170,23 @@ private bool IsOpenProperty(ODataProperty property)
         /// <param name="isTopLevel">true when writing a top-level property; false for nested properties.</param>
         /// <param name="allowStreamProperty">Should pass in true if we are writing a property of an ODataResource instance, false otherwise.
         /// Named stream properties should only be defined on ODataResource instances.</param>
+        /// <param name="omitNullValues"> Whether to omit null property values for writing.
+        /// Value should be 'false' if writing top level properties or delta payload, otherwise derived from request preference header.
+        /// </param>
         /// <param name="duplicatePropertyNameChecker">The DuplicatePropertyNameChecker to use.</param>
         [SuppressMessage("Microsoft.Maintainability", "CA1506:AvoidExcessiveClassCoupling", Justification = "Splitting the code would make the logic harder to understand; class coupling is only slightly above threshold.")]
         private void WriteProperty(
             ODataProperty property,
             IEdmStructuredType owningType,
             bool isTopLevel,
             bool allowStreamProperty,
+            bool omitNullValues,
             IDuplicatePropertyNameChecker duplicatePropertyNameChecker)
         {
             WriterValidationUtils.ValidatePropertyNotNull(property);
 
+            ODataValue value = property.ODataValue;
+
             string propertyName = property.Name;

```

Comment:
To reduce unnecessary changes, can we move this back to after writing the instance annotation on line 213?

Reply:
sure. done.

---
In reply to: [233183812](https://github.com/OData/odata.net/pull/1106#discussion_r233183812) [](ancestors = 233183812)